pkg/network/ovn_kubernetes.go

```diff
@@ -653,7 +664,7 @@ func shouldRenderIPsec(conf *operv1.OVNKubernetesConfig, bootstrapResult *bootst
 
 	// While OVN ipsec is being upgraded and IPsec MachineConfigs deployment is in progress
 	// (or) IPsec config in OVN is being disabled, then ipsec deployment is not updated.
-	renderIPsecDaemonSetAsCreateWaitOnly = isIPsecMachineConfigNotActiveOnUpgrade || (isOVNIPsecActive && !renderIPsecOVN)
+	renderIPsecDaemonSetAsCreateWaitOnly = (isIPsecMachineConfigNotActiveOnUpgrade && !renderBothIPsecDemonSetsWhenAPoolPausedState) || (isOVNIPsecActive && !renderIPsecOVN)
```

Comment:
This condition is counter-intuitive.

What about 
```
...isIPsecMachineConfigNotActiveOnUpgrade || !isIPsecMachineConfigActiveInUnPausedPools ... 
```

Also since you changed the condition, please update the comment

Reply:
the existing condition `(isIPsecMachineConfigNotActiveOnUpgrade && !renderBothIPsecDemonSetsWhenAPoolPausedState)` helps the case at which both daemonsets can be rendered without create-wait annotation, it can't be done with the suggested approach.

Reply:
So I guess you mean to say is that you want to **update** the daemonsets if machine config is active in unpaused pools and inactive in paused pools.  But I am missing the reasoning.

- Updating the deamonsets is something we didn't want to do if the machine config was not updated. Why? 
- And why can we update them in the case the pools are paused? Are both these reasonings independent?

Reply:
@jcaamano as discussed offline, updated 4.15 PR (https://github.com/openshift/cluster-network-operator/pull/2449) with following:

1. Update with both daemonsets as long as ipsec machine config is not active in any of the pool(s).
2. Get rid of checking 'paused' pool.
3. Remove `LegacyIPsecUpgrade` checks as it's not needed anymore due to update of both daemonsets at the start of upgrade itself.

Would update this PR once IPsec upgrade CI looks clean there.